src/coprocessor/endpoint.rs

```diff
@@ -420,18 +426,8 @@ impl<E: Engine> Endpoint<E> {
                 .await?;
         let latest_buckets = snapshot.ext().get_buckets();
 
-        // Check if the buckets version is latest.
-        // skip if request don't carry this bucket version.
-        if let Some(ref buckets) = latest_buckets&&
-            buckets.version > tracker.req_ctx.context.buckets_version &&
-            tracker.req_ctx.context.buckets_version!=0 {
-                let mut bucket_not_match = errorpb::BucketVersionNotMatch::default();
-                bucket_not_match.set_version(buckets.version);
-                bucket_not_match.set_keys(buckets.keys.clone().into());
-                let mut err = errorpb::Error::default();
-                err.set_bucket_version_not_match(bucket_not_match);
-                return Err(Error::Region(err));
-        }
+        check_request_key_range(&tracker.req_ctx, latest_buckets.clone())?;
```

Comment:
Shall we also return the newest bucket info if version mismatch so the client can update to the new bucket version to avoiding mismatch for a long time.

Reply:
client compare the response bucket version with the request version and then update region bucket by pd server. 
It's fine the reponse will contains bucket keys if the request verison is stale, but it will occupy the grpc message body.